test/testdata/rewriter/attr_override.rb

```diff
@@ -26,4 +28,19 @@ def initialize
 
   sig {returns(T.nilable(Integer)).override.checked(:always)}
   attr_accessor :method4
+
+  sig {returns(T.nilable(Integer)).overridable}
+  attr_accessor :method5
+
+  sig {returns(T.nilable(Integer)).abstract}
+  def method6; end
```

Comment:
This doesn't actually exercise the `castSig` code. `castSig` is a partial reimplementation of some code in `resolver/type_syntax/type_syntax.cc`, where full-fledged `sig` parsing happens. This `castSig` helper duplicates just enough of that logic to power certain rewrite passes, e.g. the `attr_accessor` one.

That means that to see the "bug," we'd need to do something like `attr_accessor :method6` here, instead of `def method6`, where it would then be observable that Sorbet treats a leading `abstract` differently from a trailing `abstract`, but it's not even clear that that's a proper "bug" because we at least report some error later in the pipeline.

[For example, sorbet.run →](https://sorbet.run/?arg=--print=rewrite-tree#%23%20typed%3A%20strict%0A%0Aclass%20A%0A%20%20extend%20T%3A%3ASig%2C%20T%3A%3AHelpers%0A%20%20abstract!%0A%0A%20%20sig%20%7Babstract.returns%28T.nilable%28Integer%29%29%7D%0A%20%20attr_accessor%20%3Amethod6%0A%0A%20%20sig%20%7Breturns%28T.nilable%28Integer%29%29.abstract%7D%0A%20%20attr_accessor%20%3Amethod7%0Aend)

It's fine to keep this here just for fun, but it doesn't exercise the same codepaths as the nearby code.

Reply:
Good to know, thanks for the explanation!